src/Utilities/Compiler/Extensions/IOperationExtensions.cs

```diff
@@ -764,6 +764,24 @@ public static bool IsSetMethodInvocation(this IPropertyReferenceOperation operat
                 _ => false,
             };
         }
+
+        public static IArgumentOperation GetArgumentForParameterAtIndex(
+            this ImmutableArray<IArgumentOperation> arguments,
+            int parameterIndex)
+        {
+            Debug.Assert(parameterIndex >= 0);
+            Debug.Assert(parameterIndex < arguments.Length);
+
+            foreach (var argument in arguments)
+            {
+                if (argument.Parameter.Ordinal == parameterIndex)
+                {
+                    return argument;
+                }
+            }
+
+            throw new NotImplementedException();
```

Comment:
I would use `InvalidOperationException` to represent invalid state and I would also include the index in the message.